### ISSUE TITLE
Getting the u32 value of a TpmHandle.

### ISSUE THREAD

**Issue by RevenantBob** [parallaxsecond/rust-tss-esapi#231]
I can't figure out how to get the u32 value of a TPM handle which is making it hard to follow the spec on querying capabilities to get the entire handle list.

It requires the previous value returned to query the next set of values. Is this an issue with the code base?

**Comment by Superhepper** [comment]
Well I think there are several ways you can get the value.

```rust
let raw_tpm_handle: TPM2_HANDLE = tpm_handle.into();
````
Is one way to do it I think. But to do what you actually want to in a better way would perhaps be.
We have a function in the tests that cleans up stuff that I think does something similar to what you want to do.

https://github.com/parallaxsecond/rust-tss-esapi/blob/56c487a101dc85e17560416d71f0fc2eb81739a6/tss-esapi/tests/context_tests/general_esys_tr_tests.rs#L20

**Comment by RevenantBob** [comment]
Thank you. I didn't realize there was TPM2_HANDLE type.

**Comment by Superhepper** [comment]
We aim to only use native rust type in the api:s. Though we do have to interact with the underlying tpm2-tss so all the native rust types should be convertible into its corresponding TPM (c-type). If there are places where you actually need to convert into the native type in order to make an operation then that type is not doing its job properly and we have to look into how to improve it.
